editor/store/effects/test/utils.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+import { registerStore, select } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { resolveSelector } from '../utils';
+
+describe( 'resolveSelector', () => {
+	it( 'Should wait for selector resolution', async () => {
+		registerStore( 'resolveStore', {
```

Comment:
Would it make to extract the store definitions as follows:
```js
	const resolveStore = {
		reducer: ( state = 'no', action ) => {
			if ( action.type === 'resolve' ) {
				return 'yes';
			}
			return state;
		},
		selectors: {
			selectAll: ( state, key ) => ( key === 'check' ) ? state : 'no-key',
		},
		resolvers: {
			selectAll: () => {
				return new Promise( ( resolve ) => {
					process.nextTick( () => resolve( { type: 'resolve' } ) );
				} );
			},
		},
	};
```
and share it between those 2 tests?